client/state/ui/notices/reducer.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import { filter } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	NEW_NOTICE,
+	REMOVE_NOTICE
+} from 'state/action-types';
+
+/**
+ * Tracks all known site objects, indexed by site ID.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function items( state = [], action ) {
```

Comment:
Should we index this by ID? Would make removing and lookup simpler and potentially more performant (avoiding the need to iterate over the array).

```jsx

export function items( state = {}, action ) {
	switch ( action.type ) {
		case NEW_NOTICE:
			state = Object.assign( {}, state, {
				[ action.notice.id ]: action.notice
			} );
			break;
		case REMOVE_NOTICE:
			state = omit( state, action.notice.id );
			break;
	}

	return state;
}
```